File: src/emitter/markdown/code-fence-lang.ts
import type { Language } from "../../cli/language.js";

export function codeFenceLang(language: Language): string {
  switch (language) {
    case "tsx":
      return "tsx";
    case "jsx":
      return "jsx";
    case "js":
      return "js";
    default:
      return "ts";
  }
}

File: crates/unsnarl-emitter-markdown/src/code_fence_lang.rs
//! Pick the code-fence language tag for the `## Input` block.
//!
//! Every recognised `Language` round-trips to its short form (`tsx`
//! / `jsx` / `js` / `ts`). `Language::Ts` is listed explicitly so
//! the default branch could be reintroduced (pointing at `"ts"`) if
//! `Language` grows new variants later without changing output for
//! existing cases.

use unsnarl_ir::language::Language;

pub fn code_fence_lang(language: Language) -> &'static str {
    match language {
        Language::Tsx => "tsx",
        Language::Jsx => "jsx",
        Language::Js => "js",
        Language::Ts => "ts",
    }
}

#[cfg(test)]
#[path = "code_fence_lang_test.rs"]
mod code_fence_lang_test;
